### PR TITLE
Schedule workflow

### PR DESCRIPTION
**Pull request by AsterNighT**
### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by AsterNighT** [comment]
I think it's ready for review. @STRRL @YangKeao

File: pkg/chaosctl/common/common.go [hunk 1]
```diff
@@ -183,7 +181,7 @@ func InitClientSet() (*ClientSet, error) {
 // GetPods returns pod list and corresponding chaos daemon
 func GetPods(ctx context.Context, chaosName string, status v1alpha1.ChaosStatus, selectorSpec v1alpha1.PodSelectorSpec, c client.Client) ([]v1.Pod, []v1.Pod, error) {
 	// get podName
-	failedMessage := status.FailedMessage
+	failedMessage := ""
 	if failedMessage != "" {
```
[the remark this answers is not in the document]
Comment (AsterNighT) [reply pkg/chaosctl/common/common.go:185]: @muse-dev ignore

File: config/crd/bases/chaos-mesh.org_schedules.yaml [hunk 2]
```diff
@@ -1898,6 +1898,7 @@ spec:
               description: 'TODO: use a custom type, as `TemplateType` contains other
                 possible values'
               type: string
+            workflow: {}
```
Comment (STRRL) [inline config/crd/bases/chaos-mesh.org_schedules.yaml:1901]: I think this is the reason that makes the CRD invalid.

**Review by STRRL — COMMENTED** [review]
Hi @AsterNighT , I try to apply that CRD into my kubernetes clusters (tried on v1.15.12 and v1.20.2), both of them told me:

```log
CustomResourceDefinition.apiextensions.k8s.io "schedules.chaos-mesh.org" is invalid: spec.validation.openAPIV3Schema.properties[spec].properties[workflow].type: Required value: must not be empty for specified object fields
```

Seems the controller-gen does NOT make the correct definition of Workflow in EmbededChaos.

FYI, then I tried to regenerate CRD YAML with controller-gen@v0.4.1 and controller-gen@v0.5.0, only controller-gen@v0.5.0 works.

**Comment by AsterNighT** [comment]
Oh my, 7162+! What the hell is that?

File: config/crd/bases/chaos-mesh.org_schedules.yaml (hunk 2 above)
Comment (AsterNighT) [reply config/crd/bases/chaos-mesh.org_schedules.yaml]: I separate schedule totally from workflow. It no longer use embeded chaos any more, but generate its own types. Does it work now?

**Review by STRRL — APPROVED** [review]
LGTM

I am sure we will refactor the definition of EmbedChaos or ScheduleItem in the future.

**Pull request merged**